src/commands/Moderation/Utilities/slowmode.ts

```diff
@@ -1,37 +1,37 @@
-import { SkyraCommand } from '@lib/structures/SkyraCommand';
+import { SkyraCommand, SkyraCommandOptions } from '@lib/structures/SkyraCommand';
 import { PermissionLevels } from '@lib/types/Enums';
 import { Time } from '@utils/constants';
 import { TextChannel } from 'discord.js';
-import { CommandStore, KlasaMessage } from 'klasa';
+import { KlasaMessage } from 'klasa';
+import { ApplyOptions } from '@skyra/decorators';
 
 const MAXIMUM_TIME = Time.Hour * 6;
 
+@ApplyOptions<SkyraCommandOptions>({
+	aliases: ['sm'],
+	bucket: 2,
+	cooldown: 5,
+	cooldownLevel: 'channel',
+	description: language => language.tget('COMMAND_SLOWMODE_DESCRIPTION'),
+	extendedHelp: language => language.tget('COMMAND_SLOWMODE_EXTENDED'),
+	permissionLevel: PermissionLevels.Moderator,
+	requiredPermissions: ['MANAGE_CHANNELS'],
+	runIn: ['text'],
+	usage: '<reset|off|seconds:integer|cooldown:cooldown>'
+})
 export default class extends SkyraCommand {
 
-	public constructor(store: CommandStore, file: string[], directory: string) {
-		super(store, file, directory, {
-			aliases: ['sm'],
-			bucket: 2,
-			cooldown: 5,
-			cooldownLevel: 'channel',
-			description: language => language.tget('COMMAND_SLOWMODE_DESCRIPTION'),
-			extendedHelp: language => language.tget('COMMAND_SLOWMODE_EXTENDED'),
-			permissionLevel: PermissionLevels.Moderator,
-			requiredPermissions: ['MANAGE_CHANNELS'],
-			runIn: ['text'],
-			usage: '<reset|off|seconds:integer|cooldown:cooldown>'
-		});
-
-		this.createCustomResolver('cooldown', async (arg, possible, message) =>
-			(await this.client.arguments.get('timespan')!.run(arg, possible, message)) / 1000);
-	}
-
 	public async run(message: KlasaMessage, [cooldown]: ['reset' | 'off' | number]) {
-		if (cooldown === 'reset' || cooldown === 'off') cooldown = 0;
+		if (cooldown === 'reset' || cooldown === 'off' || cooldown < 0) cooldown = 0;
 		else if (cooldown >= MAXIMUM_TIME) throw message.language.get('COMMAND_SLOWMODE_TOO_LONG');
 		const channel = message.channel as TextChannel;
 		await channel.setRateLimitPerUser(cooldown);
 		return message.sendLocale('COMMAND_SLOWMODE_SET', [cooldown * 1000]);
 	}
 
+	public async init() {
+		this.createCustomResolver('cooldown', async (arg, possible, message) =>
+			(await this.client.arguments.get('timespan')!.run(arg, possible, message)) / 1000);
```

Comment:
```suggestion
			(await kTimespanArg.run(arg, possible, message)) / 1000);
```
and create the private field: `private kTimespanArg = this.client.arguments.get('timespan')!`.
That way the look up in the store is cached...

Reply:
Actually, we should not define other pieces instances as a property, because of raceconditions and that we'd need to reload this command when we reload timespan.